### ISSUE TITLE
KeyError: 'sort'

### ISSUE THREAD

**Issue by sanicki** [hjalp/leddit#1]
Thanks for releasing this. I'm trying to set it up for a very small subreddit with 4.4k subscribers until Lemmy adoption will support that kind of niche community.

When I try to run it I am getting the following "KeyError":

```
Traceback (most recent call last):
  File "/home/ubuntu/leddit/src/main.py", line 74, in <module>
    syncer.scrape_new_posts()
  File "/home/ubuntu/leddit/src/utils/syncer.py", line 32, in scrape_new_posts
    sort = com['sort']
           ~~~^^^^^^^^
KeyError: 'sort'
```

**Comment by hjalp** [comment]
Hey, thanks for trying out the bot! What does the `community_map` part of your `config.yaml` file look like? I realize that the tooltip saying that "`sort` defaults to new if not provided" might imply that it can be removed entirely from the `community_map`, although what it actually means is that `sort` can be left blank. The code will look for it and crash if it's not there at all.

I think it also defaults to `hot` rather than `new` right now. I'll add a check for missing items in the config that will print clear error messages instead of this.

This is what the `community_map` should look like:
```yaml
community_map:
  - subreddit: subreddit_name
    community: lemmy_community
    sort: new
    post_header: '##### This is an automated archive made by the [Leddit Bot](https://github.com/hjalp/leddit).'
```

**Comment by sanicki** [comment]
My modified `community_map` in `config.yaml` within the `leddit` folder does include `sort: new`.

I'm running the script from within `src` because running it from `leddit` produces:

`FileNotFoundError: [Errno 2] No such file or directory: './data/config.yaml'`

Mentioning in case it expects something somewhere else.

**Comment by hjalp** [comment]
It expects `config.yaml` to be located in the folder `leddit/src/data` (which I now realize does not appear on GitHub because all of its files, which contain user data, are ignored by Git) and to be run from the `src` folder. You're right that `main.py` should be run from within `src` or it won't be able to find the config or database file.

**Comment by sanicki** [comment]
Copied config to `leddit/src/data/config.yml` and ran `main.py` from `src`. Got further. Now receiving:

`TypeError: 'NoneType' object is not subscriptable`

It looks like it added a post from "1 month ago" (according to Reddit).

The replies were not added. All of the replies to the post also appear as "1 month ago" (according to Reddit.)

**Comment by hjalp** [comment]
That's strange, the age of the post should not be an issue at all. I tested the bot on a subreddit where the most recent post was from 2021 and it was cloned successfully. Are there more details around the `NoneType` error that show what function/line the error was generated in?

And if you don't mind, you could also share the subreddit's name so that I can try to fetch posts from it myself and see what is going wrong with the bot. If you don't want to reveal the subreddit publicly, you could DM me on Lemmy at @Andreas@leddit.danmark.party.

**Comment by hjalp** [comment]
Thank you so much for bringing this up. This crash happened because that section of code in the error message has no error handling and assumes that there is always at least one comment in the database, which means that first-time runs of the bot will always fail.

Fixed in e6448f74d0d503b868eac23cd4fb04b3cdda7124

**Comment by sanicki** [comment]
This appears to be fixed. Thank you!
